src/unstract/sdk/utils/common_utils.py

```diff
@@ -54,3 +55,35 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def capture_metrics(func):
+    """Decorator to capture metrics at the start and end of a function."""
+
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
```

Comment:
NIT: It'll be safer to check if `self` contains the attributes `_run_id` / `_capture_metrics` and `_metrics` beforehand to avoid any errors. If this decorator is used in a class which doesn't have these attributes, it can result in an error during the collection process